controller/core/pattern_selector.py

```diff
@@ -112,10 +158,19 @@ async def launchpadListener(self):
             self.launchpad.mode = lpminimk3.Mode.PROG  # Switch to the programmer mode
         else:
             print(f"No launchpad found.")
-            return
-
-        for button in self.launchpad.panel.buttons():
-            button.led.color = self._LED_COLOR_INACTIVE
+        if self.launchpad:
+            for button in self.launchpad.panel.buttons():
+                button.led.color = self._LED_COLOR_INACTIVE
+        
+        #Check for connected DMX controller
+        try:
+            self.dmx = serial.Serial(DMX_ADDR, baudrate=57600, stopbits=2)
```

Comment:
I'm surprised baudrate=57600 works! I thought DMX always runs at 250Kbaud

Reply:
Same! I was using 250k to begin with, but got 57600 from here:

https://github.com/SavinaRoja/DMXEnttecPro/blob/master/src/DMXEnttecPro/controller.py

Doesn't really make sense to me tbh, but it works! Subject for further experimentation

Reply:
Does this code work with both of your Dmx dongles? It could be that the Enttec device uses a separate baudrate and protocol for the USB interface and internally creates/consumes DMX signals on the XLR side at 250k. This would also explain the comment I made about the start sequence.

Reply:
Good question, ran out of time before I got to test that. Config will be different, but I wonder if that's the only thing that changes -- if so, next question is why the Enttec box is $150 more expensive...